components/ai_chat/core/engine/remote_completion_client.cc

```diff
@@ -116,17 +113,26 @@ const GURL GetEndpointBaseUrl() {
 RemoteCompletionClient::RemoteCompletionClient(
     const std::string& model_name,
     const base::flat_set<std::string_view>& stop_sequences,
-    scoped_refptr<network::SharedURLLoaderFactory> url_loader_factory)
+    scoped_refptr<network::SharedURLLoaderFactory> url_loader_factory,
+    AIChatCredentialManager* credential_manager)
     : model_name_(model_name),
       stop_sequences_(stop_sequences),
-      api_request_helper_(GetNetworkTrafficAnnotationTag(),
-                          url_loader_factory) {
+      api_request_helper_(GetNetworkTrafficAnnotationTag(), url_loader_factory),
+      credential_manager_(credential_manager) {
   // Validate configuration
-  const GURL api_base_url = GetEndpointBaseUrl();
+  GURL api_base_url = GetEndpointBaseUrl(false);
+  if (!api_base_url.is_empty()) {
+    // Crash quickly if we have an invalid non-empty Url configured
+    // as a build flag
+    CHECK(api_base_url.is_valid()) << "Default API Url generated was invalid."
+                                      "Please check configuration parameter.";
+  }
+
+  api_base_url = GetEndpointBaseUrl(true);
   if (!api_base_url.is_empty()) {
     // Crash quickly if we have an invalid non-empty Url configured
     // as a build flag
-    CHECK(api_base_url.is_valid()) << "API Url generated was invalid."
+    CHECK(api_base_url.is_valid()) << "Premium API Url generated was invalid."
```

Comment:
I don't think we need these checks on startup anymore - the URL will always be valid.